js/main.js

```diff
@@ -155,6 +157,42 @@ function onPencilClick(event) {
   $photoUrl.value = data.editing.entryURL;
   $titleInput.value = data.editing.entryTitle;
   $notesInput.value = data.editing.entryNotes;
+  $deleteLink.className = 'deleteLink';
 }
 
 $entryList.addEventListener('click', onPencilClick);
+
+var $modal = document.querySelector('.overlay');
+var $cancelBtn = document.querySelector('.cancel-btn');
+var $confirmBtn = document.querySelector('.confirm-btn');
+
+function onDeleteClick(event) {
+  $modal.className = 'overlay flex align-center';
+}
+
+$deleteLink.addEventListener('click', onDeleteClick);
+
+function onCancelClick(event) {
+  $modal.className = 'hidden';
+}
+
+$cancelBtn.addEventListener('click', onCancelClick);
+
+function onConfirmDeleteClick(event) {
+  for (var entryNum = 0; entryNum < data.entries.length; entryNum++) {
+    if (data.editing.entryId === data.entries[entryNum].entryId) {
+      data.entries.splice(entryNum, 1);
+    }
+  }
+  var $domEntriesList = document.querySelectorAll('.entryList li');
+  for (var domEntry = 0; domEntry < $domEntriesList.length; domEntry++) {
+    if (data.editing.entryId === parseInt($domEntriesList[domEntry].getAttribute('data-entry-id'))) {
+      $domEntriesList[domEntry].remove();
+    }
+  }
+  $modal.className = 'hidden';
+  data.nextEntryId--;
```

Comment:
This shouldn't be done, just because an entry is deleted doesn't mean that the next id moves backwards.